src/services/carrier_account_service.js

```diff
@@ -40,13 +40,21 @@ export default (easypostClient) =>
      * @returns {CarrierAccount} - The updated carrier account.
      */
     static async update(id, params) {
-      const url = `carrier_accounts/${id}`;
-      const wrappedParams = {
-        carrier_account: params,
-      };
+      const carrierAccount = await this.retrieve(id);
+
+      if (!carrierAccount) {
+        throw new InvalidRequestError({
+          message: util.format(Constants.NO_OBJECT_FOUND, 'carrier account'),
+        });
+      }
```

Comment:
We don't need this, our API will handle this when the carrier account doesn't exist

Reply:
Yes, but this cancels the function early if the carrier account doesn't exist.

Reply:
Agreed, let's not add extra errors where they aren't needed. Our API handles these well. By introducing our own errors locally, we have to maintain them, we start to splinter the experience across libs when one works one way and another, another. It also means that the actual error messages users expect may differ since our lib message will be different from the API message on a 404 type error.

Reply:
I see, misunderstanding on my part. If the API responds with a 4xx when attempting that, we're already set up to throw an exception, no need to double-catch. Got it, will remove.